misk-events/src/main/kotlin/misk/events/Producer.kt

```diff
@@ -17,12 +17,18 @@ interface Producer {
    *
    * Transactions remaining outstanding until the application calls commit, or rollback, or until
    * a producer specific timeout occurs.
+   *
+   * Transactions may only be used from the thread that began them; accessing a transaction
+   * in multiple threads or handing it off between threads has undefined behavior.
+   *
+   * Having multiple outstanding transactions on the same thread also has undefined behavior.
```

Comment:
is this something we can/want to actually restrict out-right (similar to how we attempt to block nested db transactions)